### PR TITLE
Paginated request

### PR DESCRIPTION
**Pull request by jaysonlarin**
Using paginated_request, this will return the first 2000 records of an endpoint.
Maximum of data returned is 2000.

limit paginated_request calls on this endpoints: 
  - biometrics, diabetes, fitness, nutrition, routine, sleep, tobacco_cessation, weight

using weightwatchers as my org
```
client.get_routine - ordinary call - 100 records
#<Benchmark::Tms:0x007fc4418a1b20 @label="", @real=2.434471, @cstime=0.0, @cutime=0.0, @stime=0.0, @utime=0.020000000000000018, @total=0.020000000000000018>
client.get_routine(paginated: "true") - paginated_request - 2000 records
#<Benchmark::Tms:0x007fc4420068e8 @label="", @real=9.464795, @cstime=0.0, @cutime=0.0, @stime=0.020000000000000018, @utime=0.13, @total=0.15000000000000002>
```

### CODE REVIEW

**Comment by markphelps** [comment]
Is anyone asking for this feature?

why not just expose the ability to add the paginated = true for each of the endpoints and have the client consumer handle the pagination? it seems strange for our API client to handle pagination up to 2k records, what if I want 5k?

tl;dr I dont think we should handle pagination in the client, just expose the ability to make the call and have the consumers of our client handle the collection of multiple pages.

**Comment by jaysonlarin** [comment]
@markphelps thanks for the review. There is an initial effort in the direction you were pertaining to. Correct me if i am wrong but is it something like this PR https://github.com/Validic/validic/pull/29?

**Comment by markphelps** [comment]
@jaysonlarin @jjlangholtz @allcentury yes, what I'm saying is its not normal behavior for a client to do default paging. normally you just expose the ability to page (which we are already doing). 

We could perhaps add some public methods such as `next/previous` which would allow for more natural paging of the results set, however this would require that the results would be an object instead of a hash.

tl;dr: It's not good practice to do paging for the user because we have no idea how many pages they may want.. we should allow them to page on their own (which we already do)
